pkg/kubernetes/pod/v1alpha1/kubernetes.go

```diff
@@ -201,3 +264,68 @@ func (k *KubeClient) GetRaw(name string, opts metav1.GetOptions) ([]byte, error)
 	}
 	return json.Marshal(p)
 }
+
+// ExecOutput struct contains stdout and stderr
+type ExecOutput struct {
+	Stdout string `json:"stdout"`
+	Stderr string `json:"stderr"`
+}
+
+// Exec runs a command remotely in a container of a pod
+func (k *KubeClient) Exec(name string,
+	opts *corev1.PodExecOptions) (*ExecOutput, error) {
+	cli, err := k.getClientsetOrCached()
+	if err != nil {
+		return nil, err
+	}
+	config, err := k.getKubeConfigOrCached()
+	if err != nil {
+		return nil, err
+	}
+	return k.exec(cli, config, name, k.namespace, opts)
+}
+
+// ExecRaw runs a command remotely in a container of a pod
+// and returns raw output
+func (k *KubeClient) ExecRaw(name string,
+	opts *corev1.PodExecOptions) ([]byte, error) {
+	execOutput, err := k.Exec(name, opts)
+	if err != nil {
+		return nil, err
+	}
+	return json.Marshal(execOutput)
+}
+
+// defaultExec is the default implementation of execFn
+func (k *KubeClient) defaultExec(cli *clientset.Clientset, config *rest.Config, name, namespace string,
```

Comment:
Please use this as a function. No need to tie this up as method of KubeClient.

In addition, please move this function to the top, just after execFn is defined

Reply:
Done